iceoryx_utils/include/iceoryx_utils/cxx/expected.hpp

```diff
@@ -26,6 +26,44 @@ namespace iox
 {
 namespace cxx
 {
+namespace internal
+{
+/// @brief Type trait which verifies whether the passed type T has INVALID_STATE
+///        Overload chosen when INVALID_STATE is not present
+template <typename T, typename = void>
+struct has_invalid_state_member : std::false_type
+{
+};
+
+/// @brief Type trait which verifies whether the passed type T has INVALID_STATE
+///        Overload chosen when INVALID_STATE is present
+template <typename T>
+struct has_invalid_state_member<T, std::void_t<decltype(T::INVALID_STATE)>> : std::true_type
+{
+};
+
+template <typename... T>
+struct is_optional : std::false_type
```

Comment:
Could you please rename this into `IsOptional`, declare here only the forward declaration and move the template implementation into the inl file.

Reply:
Done